validator/app.py

```diff
@@ -33,13 +33,17 @@ def create_app(**kwargs):
     except FileNotFoundError:
         raise FileNotFoundError("Bad or no DATA_DIR defined")
 
-    df_innovation_, df_domain_, df_questions_ = get_fixed_data(data_dir)
+    df_innovation_, df_domain_, df_questions_, feature_weights, default_feature_weight_id = get_fixed_data(data_dir)
+    feature_weights_key = app.config.get("DEFAULT_FEATURE_WEIGHTS_KEY")
+    if feature_weights_key not in feature_weights:
```

Comment:
How can feature_weights_key be not in feature_weights?

Reply:
It's coming from a setting, and it's looking up in a data store that came from elsewhere - it's possible they could get out of sync. Note that there's a bigger problem with this code: it's returning default_feature_weight_id and ignoring it. I think for your refactor, that's correct: we will return it as part of the feature_weights dict. But we should use that one, and only fallback to DEFAULT_FEATURE_WEIGHTS_KEY if there are no feature_weights loaded, or if they have no default key (or of that key is not in the dict)